sim/src/proto/valueType.cpp

```diff
@@ -0,0 +1,124 @@
+#include <proto/valueType.h>
+
+namespace proto {
+
+std::string valStoredTypeName(const bento::protos::Value_Primitive& val) {
+    std::string name = "Primitive[";
+    switch (val.value_case()) {
+        case bento::protos::Value_Primitive::kInt8:
+            name += "8-bit Integer";
+            break;
+        case bento::protos::Value_Primitive::kInt32:
+            name += "32-bit Integer";
+            break;
+        case bento::protos::Value_Primitive::kInt64:
+            name += "64-bit Integer";
+            break;
+        case bento::protos::Value_Primitive::kFloat32:
+            name += "32-bit Float";
+            break;
+        case bento::protos::Value_Primitive::kFloat64:
+            name += "64-bit Float";
+            break;
+        case bento::protos::Value_Primitive::kStrVal:
+            name += "String";
+            break;
+        case bento::protos::Value_Primitive::kBoolean:
+            name += "Boolean";
+            break;
+        case bento::protos::Value_Primitive::VALUE_NOT_SET:
+            name += "None";
+            break;
+        default:
+            name += "UNHANDLED TYPE";
```

Comment:
Might be possible to extract the name of the oneof field  from the `Value_Primitive`'s descriptor:
https://github.com/protocolbuffers/protobuf/issues/4215#issuecomment-359991115

Reply:
Resolved in 1eb74c4